pkg/controller/seed-controller-manager/kubernetes/launching.go

```diff
@@ -40,3 +47,39 @@ func (r *Reconciler) clusterIsReachable(ctx context.Context, c *kubermaticv1.Clu
 
 	return true, nil
 }
+
+func (r *Reconciler) etcdUseStrictTLS(ctx context.Context, c *kubermaticv1.Cluster) (bool, error) {
+	statefulSet := &appsv1.StatefulSet{}
+	err := r.Client.Get(ctx, types.NamespacedName{Namespace: c.Status.NamespaceName, Name: resources.EtcdStatefulSetName}, statefulSet)
+
+	if err != nil {
+		// if the StatefulSet for etcd doesn't exist yet, a new one can be deployed with strict TLS peers
+		if kerrors.IsNotFound(err) {
+			return true, nil
+		} else {
+			return false, err
+		}
+	}
+
+	pods := &corev1.PodList{}
+	labelSet := etcd.GetBasePodLabels(c)
+
+	err = r.Client.List(ctx, pods, &client.ListOptions{
+		Namespace:     c.Status.NamespaceName,
+		LabelSelector: labels.SelectorFromSet(labelSet),
+	})
+
+	if err != nil {
+		return false, err
+	}
+
+	useStrictTls := true
+
+	for _, pod := range pods.Items {
+		if _, ok := pod.Annotations[resources.EtcdTLSEnabledAnnotation]; !ok {
+			useStrictTls = false
```

Comment:
Should we break here or do we need to check the rest of the list?

Reply:
You're right, I don't think we need to. We could even `return false, nil` in here and get rid of `useStrictTls` completely by returning `true, nil` in the last line of this function.